guide/api-design-guidelines.md

```diff
@@ -0,0 +1,193 @@
+model SecretReference {
+  @doc("The ID of an Applications.Core/SecretStore resource containing sensitive data.")
+  source: string;
+
+  @doc("The key for the secret in the secret store.")
+  key: string;
+}
+
+```        
+
+#### Usage of SecretReference
+
+<a href="#secret-envvar" name="secret-envvar">:white_check_mark:</a> **DO** use the `SecretReference` type to reference a single (scalar) secret value in a resource property. Resource properties should reference a `Applications.Core/secretStores` instead of directly containing secret data.
+
+This pattern simplifies the overall design of Radius by reducing the number of places where we store secret data. 
+The structure also allows for environment variables to refer to other resources such as `ConfigMaps`, `Pod Fields` etc. in the future. 
+Examples include cases where environment variables containing sensitive information are injected into a container or used in a Terraform execution process.
+
+#### Example TypeSpec Definition
+
+```tsp
+
+@doc("environment")
+env?: Record<EnvironmentVariable>;
+
+@doc("Environment variables type")
+model EnvironmentVariable {
+
+  @doc("The value of the environment variable")
+  value?: string;
+
+  @doc("The reference to the variable")
+  valueFrom?: EnvironmentVariableReference;
+}
+
+@doc("The reference to the variable")
+model EnvironmentVariableReference {
+  @doc("The secret reference")
+  secretRef: SecretReference;
+}
+
+```
+
+#### Example Bicep Definition
+
+```bicep
+
+env: {
+  DB_USER: { value: 'DB_USER' }
+  DB_PASSWORD: {
+    valueFrom: {
+      secretRef: {
+        source: secret.id
+        key: 'DB_PASSWORD'
+      }
+    } 
+  }
+} 
+
+```
+
+### :no_entry: DO NOT: Store sensitive data as clear text
+Sensitive data MUST NEVER be stored directly in resource definitions or configuration files as clear text. Instead, use references to secrets stored in secret stores.
+
+#### Example of What NOT to Do
+
+```bicep
+
+env: {
+  DB_USER: { value: 'DB_USER' }
+  DB_PASSWORD:  { value: 'myPlainTextPassword' }  // ❌ This is NOT allowed
+} 
+
+```
+
+### SecretConfig Model
+
+<a href="#secretconfig-model" name="secretconfig-model">:white_check_mark:</a> **DO** implement this model when the structure of the secret store resource is known and a component or resource in Radius requires authentication to external systems, such as private container registries, TLS certificates.
+
+```tsp
+
+@doc("Secret Configuration used to authenticate to external systems.")
+model SecretConfig {
+  @doc("The ID of an Applications.Core/secretStores resource containing credential information.")
+  secret?: string;
+}
+
+```        
+
+#### Usage of SecretConfig
+
+<a href="#secretconfig-ext" name="secretconfig-ext">:white_check_mark:</a> **DO** use the SecretConfig type to reference a structured set of secret values in a resource property. Resource properties should reference a Applications.Core/secretStores instead of directly containing secret data.
```

Comment:
```suggestion
<a href="#secretconfig-ext" name="secretconfig-ext">:white_check_mark:</a> **DO** use the `SecretConfig` type to reference a structured set of secret values in a resource property. Resource properties should reference a `Applications.Core/secretStores` instead of directly containing secret data.
```